index.bs

```diff
@@ -1428,23 +1403,113 @@ session.CapabilityRequest = {
   ? browserName: text,
   ? browserVersion: text,
   ? platformName: text,
-  ? proxy: {
-    ? proxyType: "pac" / "direct" / "autodetect" / "system" / "manual",
-    ? proxyAutoconfigUrl: text,
-    ? ftpProxy: text,
-    ? httpProxy: text,
-    ? noProxy: [*text],
-    ? sslProxy: text,
-    ? socksProxy: text,
-    ? socksVersion: 0..255,
-  },
+  ? proxy: session.ProxyConfiguration,
+  ? webSocketUrl: bool,
   Extensible
 };
 </pre>
 
 The <code>session.CapabilityRequest</code> type represents a specific set of
 requested capabilities.
 
+WebDriver BiDi defines [=additional WebDriver capability|additional WebDriver
+capabilities=]. The following tables enumerates the capabilities each
+implementation must support for WebDriver BiDi.
+
+<pre class=simpledef>
+Capability: <dfn export>WebSocket URL</dfn>
+Key: "<code>webSocketUrl</code>"
+Value type: boolean
+Description: Defines the current session's support for bidirection connection.
+</pre>
+
+<div algorithm="webSocketUrl capability deserialization algorithm">
+The [=additional capability deserialization algorithm=] for the
+"<code>webSocketUrl</code>" capability, with parameter |value| is:
+
+1. If |value| is not a boolean, return [=error=] with [=error code|code=]
+   [=invalid argument=].
+
+1. Return [=success=] with data |value|.
+
+</div>
+
+<div algorithm="webSocketUrl capability serialization algorithm">
+The [=matched capability serialization algorithm=] for the "<code>webSocketUrl</code>" capability,
+with parameter |value| is:
+
+1. If |value| is false, return [=success=] with data null.
+
+1. Return [=success=] with data true.
+
+</div>
+
+#### The session.ProxyConfiguration Type #### {#type-session-ProxyConfiguration}
+
+[=remote end definition=] and [=local end definition=]
+
+<pre class="cddl remote-cddl local-cddl">
+session.ProxyConfiguration = {
+   session.AutodetectProxyConfiguration //
+   session.DirectProxyConfiguration //
+   session.ManualProxyConfiguration //
+   session.PacProxyConfiguration //
+   session.SystemProxyConfiguration //
+}
+
+session.AutodetectProxyConfiguration = (
+   ; Indicates that the proxy to use should be detected in an
+   ; implementation-specific way.
+   proxyType: "autodetect"
+   Extensible
```

Comment:
I thought we decided at some point that `Extensible` didn't really do anything?

Reply:
1. "With Extensible" => "The object may have other properties"
2. "Without Extensible" => "The object may or may not have other properties"

For (2), it's up to the implementation to define semantics there because CDDL is not explicit.

Reply:
CDDL has an extension mechanism using `//=`. This allows anyone to extend CDDL without tampering with the original code, so in practice, we can remove Extensible without repercussions.

I'll still keep it here, but we can discuss in another PR whether we should remove all `Extensible`s.